base/linalg/tridiag.jl

```diff
@@ -312,7 +321,19 @@ immutable Tridiagonal{T} <: AbstractMatrix{T}
     d::Vector{T}     # diagonal
     du::Vector{T}    # sup-diagonal
     du2::Vector{T}   # supsup-diagonal for pivoting
+
+    Tridiagonal(dl::Vector{T}, d::Vector{T}, du::Vector{T}, du2::Vector{T}) = new(dl, d, du, du2)
+    function Tridiagonal(n::Integer, m::Integer)
+        if n != m
+            throw(DimensionMismatch("Tridiagonal matrix must be square"))
+        end
+        new(Vector{T}(n-1), Vector{T}(n), Vector{T}(n-1), zeros(T, n-2))
```

Comment:
That's correct. The LU factorization requires an extra super diagonal so it's allocated it during on the construction of the matrix since factorization/linear solve is only supported methods for tridiagonals in LAPACK anyway. In that way, I could reuse the same `LU` type for both dense and tridiagonal `LU`. If `du2` was allocated during the factorization, then it would require a new  `TridiagonalLU` with an extra field.